src/distributions/microsoft/installer.ts

```diff
@@ -0,0 +1,122 @@
+import { JavaBase } from '../base-installer';
+import { JavaDownloadRelease, JavaInstallerOptions, JavaInstallerResults } from '../base-models';
+import semver from 'semver';
+import { extractJdkFile, getDownloadArchiveExtension, isVersionSatisfies } from '../../util';
+import * as core from '@actions/core';
+import { MicrosoftVersion, PlatformOptions } from './models';
+import * as tc from '@actions/tool-cache';
+import fs from 'fs';
+import path from 'path';
+
+const supportedPlatform = `'linux', 'macos', 'windows'`;
+
+const supportedArchitecture = `'x64', 'armv7', 'aarch64'`;
+
+export class MicrosoftDistributions extends JavaBase {
+  constructor(installerOptions: JavaInstallerOptions) {
+    super('Microsoft', installerOptions);
+  }
+
+  protected async downloadTool(javaRelease: JavaDownloadRelease): Promise<JavaInstallerResults> {
+    core.info(
+      `Downloading Java ${javaRelease.version} (${this.distribution}) from ${javaRelease.url} ...`
+    );
+    const javaArchivePath = await tc.downloadTool(javaRelease.url);
+
+    core.info(`Extracting Java archive...`);
+    const extension = getDownloadArchiveExtension();
+    const extractedJavaPath = await extractJdkFile(javaArchivePath, extension);
+
+    const archiveName = fs.readdirSync(extractedJavaPath)[0];
+    const archivePath = path.join(extractedJavaPath, archiveName);
+
+    const javaPath = await tc.cacheDir(
+      archivePath,
+      this.toolcacheFolderName,
+      this.getToolcacheVersionName(javaRelease.version),
+      this.architecture
+    );
+
+    return { version: javaRelease.version, path: javaPath };
+  }
+
+  protected async findPackageForDownload(range: string): Promise<JavaDownloadRelease> {
+    if (this.architecture !== 'x64' && this.architecture != 'aarch64') {
+      throw new Error(`Unsupported architecture: ${this.architecture}`);
+    }
+    const availableVersionsRaw = await this.getAvailableVersions();
+
+    const opts = this.getPlatformOption();
+    const availableVersions = availableVersionsRaw.map(item => ({
+      url: `https://aka.ms/download-jdk/microsoft-jdk-${item.fullVersion}-${opts.os}-${this.architecture}.${opts.archive}`,
+      version: this.convertVersionToSemver(item)
+    }));
+
+    const satisfiedVersion = availableVersions
+      .filter(item => isVersionSatisfies(range, item.version))
+      .sort((a, b) => -semver.compareBuild(a.version, b.version))[0];
+
+    if (!satisfiedVersion) {
+      const availableOptions = availableVersions.map(item => item.version).join(', ');
+      const availableOptionsMessage = availableOptions
+        ? `\nAvailable versions: ${availableOptions}`
+        : '';
+      throw new Error(
+        `Could not find satisfied version for semver ${range}. ${availableOptionsMessage}`
```

Comment:
```suggestion
        `Could not find satisfied version for SemVer ${range}. ${availableOptionsMessage}`
```

I see this error message both ways in the codebase, but let's use the official stylization for new code